packages/atomic/src/components/search-box-suggestions/suggestions-common.ts

```diff
@@ -0,0 +1,41 @@
+import {SearchBox} from '@coveo/headless';
+import {VNode} from '@stencil/core';
+import {buildCustomEvent} from '../../utils/event-utils';
+import {Bindings} from '../../utils/initialization-utils';
+
+export interface SearchBoxSuggestionElement {
+  value: string;
+  onClick(): void;
+  content: VNode;
+}
+
+export interface SearchBoxSuggestions {
+  // TODO: add query context onInput
+  // TODO: add position (priority in the list)
+  onInput(): Promise<unknown> | void;
+  renderItems(): SearchBoxSuggestionElement[];
+}
+
+export type SearchBoxSuggestionsEvent = (
+  bindings: SearchBoxSuggestionsBindings
+) => SearchBoxSuggestions;
+
+export interface SearchBoxSuggestionsBindings extends Bindings {
+  id: string;
+  searchBoxController: SearchBox;
+  numberOfQueries: number;
+}
+
+export const dispatchSearchBoxSuggestionsEvent = (
+  event: SearchBoxSuggestionsEvent,
+  element: Element
+) => {
+  const canceled = element.dispatchEvent(
+    buildCustomEvent('atomic/searchBoxSuggestion', event)
```

Comment:
Would the event's role be clearer if it included `/register` or similar at the end? e.g.

```
atomic/searchBoxSuggestion/register
```